zetaclient/config/types.go

```diff
@@ -17,93 +19,196 @@ type ClientConfiguration struct {
-	InTxTicker                  uint64
-	OutTxTicker                 uint64
-	WatchUTXOTicker             uint64
-	ConfCount                   uint64
-	ConnectorContractAddress    string
-	ZETATokenContractAddress    string
-	ERC20CustodyContractAddress string
-	OutboundTxScheduleInterval  int64
-	OutboundTxScheduleLookahead int64
-}
-
-func NewCoreParams() *CoreParams {
-	return &CoreParams{
-		ChainID:                     0,
-		GasPriceTicker:              0,
-		InTxTicker:                  5,
-		OutTxTicker:                 3,
-		WatchUTXOTicker:             5,
-		ConfCount:                   0,
-		ConnectorContractAddress:    "",
-		ZETATokenContractAddress:    "",
-		ERC20CustodyContractAddress: "",
-	}
-}
-
 type EVMConfig struct {
-	Client     *ethclient.Client
-	Chain      common.Chain
-	Endpoint   string
-	CoreParams *CoreParams
+	ostypes.CoreParams
+	Chain    common.Chain
+	Endpoint string
 }
 
 type BTCConfig struct {
+	ostypes.CoreParams
+
 	// the following are rpcclient ConnConfig fields
 	RPCUsername string
 	RPCPassword string
 	RPCHost     string
 	RPCParams   string // "regtest", "mainnet", "testnet3"
-
-	CoreParams *CoreParams
 }
 
 type Config struct {
-	Peer          string
-	PublicIP      string
-	LogFormat     string
-	LogLevel      zerolog.Level
-	LogSampler    bool
-	PreParamsPath string
-	Keygen        zetaObserverTypes.Keygen
-	ChainID       string
-	ZetaCoreURL   string
-	AuthzGranter  string
-	AuthzHotkey   string
-
-	ChainsEnabled       []common.Chain
-	EVMChainConfigs     map[int64]*EVMConfig // TODO : chain to chain id
-	BitcoinConfig       *BTCConfig
-	P2PDiagnostic       bool
-	ConfigUpdateTicker  uint64
-	P2PDiagnosticTicker uint64
-	TssPath             string
-	TestTssKeysign      bool
+	Peer                string        `json:"Peer"`
+	PublicIP            string        `json:"PublicIP"`
+	LogFormat           string        `json:"LogFormat"`
+	LogLevel            zerolog.Level `json:"LogLevel"`
+	LogSampler          bool          `json:"LogSampler"`
+	PreParamsPath       string        `json:"PreParamsPath"`
+	ChainID             string        `json:"ChainID"`
+	ZetaCoreURL         string        `json:"ZetaCoreURL"`
+	AuthzGranter        string        `json:"AuthzGranter"`
+	AuthzHotkey         string        `json:"AuthzHotkey"`
+	P2PDiagnostic       bool          `json:"P2PDiagnostic"`
+	ConfigUpdateTicker  uint64        `json:"ConfigUpdateTicker"`
+	P2PDiagnosticTicker uint64        `json:"P2PDiagnosticTicker"`
+	TssPath             string        `json:"TssPath"`
+	TestTssKeysign      bool          `json:"TestTssKeysign"`
+
+	// chain specific fields are updatable at runtime and shared across threads
+	cfgLock         *sync.RWMutex        `json:"-"`
+	Keygen          ostypes.Keygen       `json:"Keygen"`
+	ChainsEnabled   []common.Chain       `json:"ChainsEnabled"`
+	EVMChainConfigs map[int64]*EVMConfig `json:"EVMChainConfigs"`
+	BitcoinConfig   *BTCConfig           `json:"BitcoinConfig"`
```

Comment:
JSON fields should be in snake case

Reply:
Agree on that. It would be better to do it in future PR as I'm afraid of modifying existing config file being used now. I put it in track here: https://github.com/zeta-chain/node/issues/971